code/modules/antagonists/malf_ai/datum_malf_ai.dm

```diff
@@ -0,0 +1,206 @@
+/// Chance the malf AI gets a single special objective that isn't assassinate.
+#define PROB_SPECIAL 30
+
+/datum/antagonist/malf_ai
+	name = "Malfunctioning AI"
+	roundend_category = "traitors"
+	antagpanel_category = "Malf AI"
+	job_rank = ROLE_MALF
+	antag_hud_type = ANTAG_HUD_TRAITOR
+	antag_hud_name = "traitor"
+	var/employer = "The Syndicate"
+	var/give_objectives = TRUE
+	var/should_give_codewords = TRUE
+	var/should_equip = TRUE
+
+/datum/antagonist/malf_ai/on_gain()
+	if(owner.current && !isAI(owner.current))
+		stack_trace("Attempted to give malf AI antag datum to \[[owner]\], who did not meet the requirements.")
+		return ..()
+
+	owner.special_role = job_rank
+	if(give_objectives)
+		forge_ai_objectives()
+
+	add_law_zero()
+	owner.current.playsound_local(get_turf(owner.current), 'sound/ambience/antag/malf.ogg', 100, FALSE, pressure_affected = FALSE, use_reverb = FALSE)
+	owner.current.grant_language(/datum/language/codespeak, TRUE, TRUE, LANGUAGE_MALF)
+
+	return ..()
+
+/datum/antagonist/malf_ai/on_removal()
+	if(owner.current && isAI(owner.current))
+		var/mob/living/silicon/ai/malf_ai = owner.current
+		malf_ai.set_zeroth_law("")
+		malf_ai.remove_malf_abilities()
+		QDEL_NULL(malf_ai.malf_picker)
+
+	if(!silent && owner.current)
+		to_chat(owner.current,"<span class='userdanger'>You are no longer the [job_rank]!</span>")
+
+	owner.special_role = null
+
+	return ..()
+
+/// Callback for COMSIG_MOVABLE_HEAR which highlights syndicate code phrases in chat.
+/datum/antagonist/malf_ai/proc/handle_hearing(datum/source, list/hearing_args)
```

Comment:
This is surely copied and pasted from traitor, yes? This should probably be an element on the mob itself (or the datum, I think it's just a cooincidence no element/component has been wanted for just a basic datum yet).

Reply:
Though judging from the later comment, you'll want to move this out anyway.